src/main/components/RequestBar.jsx

```diff
@@ -9,36 +10,50 @@ const RequestBar = (props) => {
   const [selected, setSelected] = useState(method);
   const [uri, setUri] = useState('');
 
+  // header info
+  const [headerType, setHeaderType] = useState('Authorization');
+  const [authType, setType] = useState('Bearer Token');
+  const [headerKey, setHeaderKey] = useState('');
 
   const handleChange = (e) => {
     const { name, value } = e.target;
     if (name === 'method') setSelected(value);
     else if (name === 'uri') setUri(value);
+
+    // header info
+    if (name === 'header') setHeaderType(value);
+    if (name === 'headerKey') setHeaderKey(`Bearer ${value}`);
+
+  };
+
+  const runTest = (link, sendingObj, testsClone, i) => {
+    const test = testsClone;
+    fetch(link, sendingObj)
+      .then((response) => {
+        test[i].status = response.status;
+        if (i === test.length - 1) setTests(test);
+      })
+      .catch(error => console.log(error));
   };
 
   function sendFetch(e) {
     e.preventDefault();
 
     if (SourceOrDest === 'source') {
       const sendingObj = { method: selected, mode: 'cors' };
+      if (headerType !== 'NONE') sendingObj.headers = { [headerType]: headerKey };
 
       fetch(uri, sendingObj)
         .then(res => res.json())
         .then(res => setData(res));
     } else if (SourceOrDest === 'dest') {
       const testsClone = [...tests];
       const sendingObj = { method: selected, mode: 'cors' };
-      let counter = 0;
+      if (headerType !== 'NONE') sendingObj.headers = { [headerType]: headerKey };
```

Comment:
we should pull both the
`const sendingObj = { method: selected, mode: 'cors' };`
and the 
`if (headerType !== 'NONE') sendingObj.headers = { [headerType]: headerKey };`
outside of the if-statement.  I wrote the first one, no problems on getting approved bc it works just fine, just noticed it now.